### PR TITLE
FRW-7296 Adjusted error response messages

### PR DESCRIPTION
**Pull request by olhalivitchuk**
## PR Description

https://spryker.atlassian.net/browse/FRW-6430

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: _includes/pbc/all/install-features/202311.0/install-the-data-exchange-api.md [hunk 1]
```diff
@@ -371,30 +371,30 @@ Ensure successful generation of Propel entities by verifying that they exist. Ad
...
-dynamic_entity.validation.entity_does_not_exist,"The entity could not be found in the database.",en_US
-dynamic_entity.validation.entity_does_not_exist,"Die Entität konnte in der Datenbank nicht gefunden werden.",de_DE
-dynamic_entity.validation.invalid_field_type,"Invalid data type for field: %tableAlias%.%fieldName%",en_US
-dynamic_entity.validation.invalid_field_type,"Ungültiger Datentyp: %tableAlias%.%fieldName%",de_DE
-dynamic_entity.validation.invalid_field_value,"Invalid data value for field: %fieldName%, row number: %rowNumber%. Field rules: %validationRules%",en_US
-dynamic_entity.validation.invalid_field_value,"Ungültiger Datenwert für das Feld: %fieldName%, Zeilennummer: %rowNumber%. Feldregeln: %validationRules%",de_DE
-dynamic_entity.validation.required_field_is_missing,"The required field must not be empty. Field: '%tableAlias%.%fieldName%'",en_US
-dynamic_entity.validation.required_field_is_missing,"Das erforderlich Feld darf nicht leer sein. Feld: '%tableAlias%.%fieldName%'",de_DE
+dynamic_entity.validation.persistence_failed,"Failed to persist the data for `%entityChains%`. Please verify the provided data and try again.",en_US
```
[the remark this answers is not in the document]
Comment (olhalivitchuk) [reply _includes/pbc/all/install-features/202311.0/install-the-data-exchange-api.md:374]: Sorry, I checked RPs and did not find any comments regarding renaming. Could you please support and share the link with me?

[the remark this answers is not in the document]
Comment (asmarovydlo) [reply _includes/pbc/all/install-features/202311.0/install-the-data-exchange-api.md:374]: https://github.com/spryker/spryker/pull/10861#discussion_r1528908631

File: docs/pbc/all/data-exchange/202311.0/tutorials-and-howtos/how-to-send-request-in-data-exchange-api.md [hunk 2]
```diff
@@ -302,7 +302,7 @@ For example, configure `isCreatable: false` for `iso3_code` and send the same re
 ```json
 [
   {
-    "message": "Modification of immutable field `iso3_code` is prohibited.",
+    "message": "Modification of immutable field `countries[0].iso3_code` is prohibited.",
```
Comment (asmarovydlo) [inline docs/pbc/all/data-exchange/202311.0/tutorials-and-howtos/how-to-send-request-in-data-exchange-api.md:305]: In reality, the chain looks differently (but must look like it is done in documentation), https://github.com/spryker/robotframework-suite-tests/pull/865 please create a bug and fix it.

**Review by asmarovydlo — CHANGES_REQUESTED** [review]

File: docs/pbc/all/data-exchange/202311.0/tutorials-and-howtos/how-to-send-request-in-data-exchange-api.md (hunk 2 above)
Comment (olhalivitchuk) [reply docs/pbc/all/data-exchange/202311.0/tutorials-and-howtos/how-to-send-request-in-data-exchange-api.md:305]: Do you mean putting the index in the brackets? If yes, in reality, it works the same as in documentation. But I saw this "weirdo" behavior in robot framework and I do not know how to explain that

Comment (asmarovydlo) [reply docs/pbc/all/data-exchange/202311.0/tutorials-and-howtos/how-to-send-request-in-data-exchange-api.md:305]: Ok, thanks

**Review by asmarovydlo — APPROVED** [review]

**Pull request merged**